draft-specifications/Model.md

```diff
@@ -235,50 +284,69 @@ A struct with unique, unordered string keys and values of heterogeneous type.
+>     [Symbol.toStringTag]: 'tagName',
 >     payload
 >   }
 >   ```
-> - **Python**: `Tagged(tag, value)` where `tag` is a `string` and `Tagged` is
+> - **Python**: `Tagged('tagName', value)` where `tag` is a `string` and `Tagged` is
 >   imported from `ocapn`.
 >
 > OCapN tagged data have a single value (are not variadic) [November 14, 2023](https://github.com/ocapn/ocapn/issues/94)
 >
 > What are tagged values for? https://github.com/ocapn/ocapn/issues/52
 
-## Remotable
+## Error
+
+A value capturing the reason for rejecting a delivery.
+
+> - **Guile**: to be proposed
+> - **JavaScript**: a JavaScript Error object
+> - **Python**: to be proposed
+>
+> We have not yet converged on any particular details about the modeling of
+> errors. The purpose of errors is typically to indicate that some requested
+> operation failed. The purpose of the contents of errors is to preserve and
+> convey diagnostic information, mostly to help debug problems, such as the
+> root cause of a surprising failure. This is a best-efforts obligation, for
+> which we have not yet decided either what contents are required, nor what is
+> allowed, nor what must be preserved as errors are passed from one site to
+> another. Until these details are decided, the only hard requirement is that
+> an error round trip to an error. We avoid any interpretation for now as to
+> whether it is the "same" error.
 
-A value that can receive messages.
+# Reference
 
-Remotable values can be either a local value or a remote value that forwards
-messages to a local value.
-The latter is a presence: a remotable that forwards messages to the peer where the
-message can be delivered to its corresponding local remotable.
+A value that can receive messages, either a Target or Promise.
 
-Comparing the identities of remotable values is pass-invariant, meaning that
-comparing the identities of any pair of remotable values will produce the same
-result on one peer if passed over OCapN to another peer and compared there,
-regardless of whether either value is a presence on either peer.
+## Target
 
-Although Remotables and Promises are both Capabilities for delivering messages,
-Promises do not support pass-invariant identity comparison.
+Targets represent either a local value or a remote value.
+OCapN can export references to local targets and can import references to
+remote targets.
+A local target handles deliveries and produces either a return value
+(fulfillment) or thrown error (rejection reason) for a message delivery.
+A remote target (a presence) forwards messages to its corresponding local target.
 
-> - **Guile**: a function
-> - **JavaScript**: A *far object* or a *remote presence* of that far object.
+> - **Guile**: a procedure
+> - **JavaScript**: Tentatively, *far object* (can handle message deliveries
+>   locally) or a *presence* (a token object that forwards messages to a
+>   corresponding far object).
 >   ```js
 >   {
 >     __proto__: {
```

Comment:
Need to say something about the `__proto__`. Also, the actual remotable object is an object that inherits from the object shown, possibly through multiple levels, each contributing their own methods.

Again, would be good for all such code examples to have similar enough text to help people reason by similarity and difference.

Reply:
We could also add a case to `pass-style` such that any `x` where `typeof x === 'function'` is a valid target and revise `Far` and `exo` to use that as their foundation, obviating this detail.

Reply:
I do not believe we should do that, but it is arguable. I'll  leave the reasons to a verbal discussion sometime.

Reply:
In any case, it would be a huge change that we're not likely to consider anytime soon. It would be good to have the JS examples reflect current endo reality.

Reply:
Simple (and IMO adequate) reason not to consider unmarked functions to be targets by default:

We are trying really hard to shift 
- from a local E-like hardened js programming model where every object and function boundary should be a defensive boundary
- to a local endo programming model where, in general, the defensive boundaries are target invocation boundaries, where targets are typically exos protected by interface guards.

For example, one of the reasons I find our shift towards more direct support of ts acceptable, despite typescript's unsound type system, is that to the exo programmer, the relevant type-like system for defensive exo boundaries will be the declarative interface guards, where the ts types directly inferred from these guards are both static and sound.

Admitting unmarked functions by default as targets would completely undo the progress we've made towards this coarsening of defensive boundaries.

Reply:
I’ll retract the proposition that we allow all bare JavaScript functions to be valid targets.

A problem with documenting Endo reality here is that it is not sufficient to express a target that can receive an arbitrary delivery. We could document Endo reality as a special case, but the general case ought to be a marked function, e.g.,

```js
({
  [Symbol.for('passStyle')]: 'target',
  target: (...args) => {},
})
```

Given that these non-normative sections are for examples and not per-language specifications, I find that the general case is sufficient for now, provided that you concur.

Reply:
For the record, @erights does not concur and I’ve removed the example, since no example that is sufficiently expressive to receive an arbitrary delivery yet exists for JavaScript.